components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineView.kt

```diff
@@ -24,7 +26,8 @@ import java.lang.IllegalStateException
 class GeckoEngineView @JvmOverloads constructor(
     context: Context,
     attrs: AttributeSet? = null,
-    defStyleAttr: Int = 0
+    defStyleAttr: Int = 0,
+    private val selectionActionDelegate: SelectionActionDelegate? = null
```

Comment:
Is there another way for this to be passed to GeckoView? Stringing it through the view constructors feels brittle, especially if we start adding more properties like this in the constructor down the line. Would a setter suffice?

Reply:
Hm, I'm not sure I follow.  I'm definitely unhappy about adding a dependency to `GeckoEngineView`, but imo adding a setter carries the same drawbacks as a constructor param in addition to adding an unnecessary lifecycle (pre vs post setter call).  I feel like it obfuscates the problem, rather than fixing it.  I'm open to alternatives here, but strongly against moving this to a setter.

Reply:
Ekager showed me a way we might be able to get this to the `GeckoSession` directly, so I'm going to try to get that working.